lua/entities/gmod_wire_ledtape.lua

```diff
@@ -0,0 +1,341 @@
+	-- This is a necessary evil to prevent stuttering.
+	local LocalToWorld_NoGarbage_Ents = {}
+
+	local function LocalToWorld_NoGarbage(ent, pos)
+		ent.LEDTapeVecs = ent.LEDTapeVecs or {}
+		local LEDTapeVecs = ent.LEDTapeVecs
+
+		if ent.LEDTapeLastPos == ent:GetPos() and ent.LEDTapeLastAng == ent:GetAngles() and LEDTapeVecs[pos] then
+			return LEDTapeVecs[pos]
+		end
+
+		LEDTapeVecs[pos] = ent:LocalToWorld(pos)
+		LocalToWorld_NoGarbage_Ents[ent] = true -- update positions at the end
+
+		return LEDTapeVecs[pos]
+	end
+
+	local function LocalToWorld_NoGarbage_End()
+		for ent, _ in pairs(LocalToWorld_NoGarbage_Ents) do
+			if not IsValid(ent) then continue end
+			ent.LEDTapeLastPos = ent:GetPos()
+			ent.LEDTapeLastAng = ent:GetAngles()
+		end
+		LocalToWorld_NoGarbage_Ents = {}
+	end
+
+	hook.Add("PostDrawOpaqueRenderables","LEDTapeCleanup",LocalToWorld_NoGarbage_End)
+
+	local function DrawBeams(width, scrollmul, mater, path, getColor, extravertex)
+
+		if not IsValid(path[1][1]) then return end
+
+		local scroll = 0
+		local beam   = render.AddBeam
+
+		local beam2     = extravertex and beam or function() end -- branchless programming ftw
+		local vertexnum = extravertex and 3 or 2
+
+		scrollmul = scrollmul / width -- scale this
+
+		render.SetMaterial(mater)
+		render.StartBeam(#path * vertexnum)
+
+			local node1 = path[1]
+
+			local pt1 = LocalToWorld_NoGarbage(node1[1], node1[2])
+
+			beam(pt1, width, scroll, getColor(node1))
+
+			for i = 2, #path do
+				local node2 = path[i]
+				local nodeEnt = node2[1]
+				if not IsValid(nodeEnt) then continue end
+				local nodeOffset = node2[2]
+
+				local pt2 = LocalToWorld_NoGarbage(nodeEnt, nodeOffset)
+				local distance = pt2:Distance(pt1) * scrollmul * 0.5
+
+				beam( pt1, width, scroll, getColor(node1))
+				scroll = scroll + distance
+				beam( pt2, width, scroll, getColor(node2))
+				beam2( pt2, width, scroll, getColor(node2)) -- add another point if extravertex is set, prevents some sprites from looking yucky
+
+				pt1 = pt2
+				node1 = node2
+			end
+
+			beam(pt1, width, scroll, getColor(node1))
+
+		render.EndBeam()
+		return pt1
+	end
+
+	function Wire_LEDTape.DrawShaded(width, scrollmul, mater, path)
+		return DrawBeams(width, scrollmul, mater, path, getLitNodeColor)
+	end
+
+	function Wire_LEDTape.DrawFullbright(width, scrollmul, color, mater, path, extravert)
+		return DrawBeams(width, scrollmul, mater, path, function(node) return color end, extravert)
+	end
+
+	function ENT:Initialize()
+
+		self:SharedInit()
+		self.ScrollMul = DEFAULT_SCALE
+
+		net.Start("LEDTapeData")
+			net.WriteEntity(self)
+			net.WriteBool(true) -- request full update
+		net.SendToServer()
+
+		if CLIENT then
+			local DrawShaded = Wire_LEDTape.DrawShaded
+			local DrawFullbright = Wire_LEDTape.DrawFullbright
+			hook.Add("PostDrawOpaqueRenderables", self, function()
+
+				if #self.Path < 2 then return end
```

Comment:
Either remove and add the hook whenever `#Path >= 2` or simplify to a single bool.